### ISSUE TITLE
dynamically index http.request.headers by default

### ISSUE THREAD

**Issue by simitt** [elastic/apm-server#4137]
We have had several occasions where users were asking about how to add additional fields to the default template, and where `http.request.headers` was mentioned as an example. 
While we are not using the values in the UI it might be worth adding the headers to the indexed fields by default.

**Comment by axw** [comment]
How would one automate removing indexing for headers if they wanted to opt out? Can this be done with the configuration we have today?

**Comment by axw** [comment]
I may wrong, but I think that's probably because we've historically been conservative about what we index by default. Core stuff is indexed and you have no way to opt out; extra stuff you have to opt in. I'd be a bit wary of making this change without providing a way to opt out.

I think when we have component templates we should be able to do this?

**Comment by simitt** [comment]
Yes, component templates will be the way to move forward with this. Closing the issue then for now.

**Comment by simitt** [comment]
After a quick offline discussion, turns out there was a misunderstanding regarding component templates. We could either 
* keep things as they are and leave it up to the users to add fields via component templates
* add more fields to the default index and leave it up to the users to remove fields they do not need. This option might lead to users removing fields, which might be necessary for the UI and potentially break things

**Comment by axw** [comment]
Maybe we can use runtime fields to avoid the cost of indexing while still making them searchable.

**Comment by felixbarny** [comment]
There's now a new option for mapping dynamic objects that we should consider using: [`dynamic: runtime`](https://github.com/elastic/elasticsearch/pull/65489). 

It removes the need for us or the user to explicitly add specific headers as runtime fields.
